tests/components/influxdb/test_init.py

```diff
@@ -93,57 +122,45 @@ def _get_write_api_mock_v2(mock_influx_client):
 async def test_setup_config_full(hass, mock_client, config_ext, get_write_api):
     """Test the setup with full configuration."""
     config = {
-        "influxdb": {
-            "host": "host",
-            "port": 123,
-            "database": "db",
-            "max_retries": 4,
-            "ssl": "False",
-        }
+        "host": "host",
+        "port": 123,
+        "max_retries": 4,
+        "ssl": True,
     }
-    config["influxdb"].update(config_ext)
+    config.update(config_ext)
 
-    assert await async_setup_component(hass, influxdb.DOMAIN, config)
-    await hass.async_block_till_done()
-    assert hass.bus.listen.called
-    assert EVENT_STATE_CHANGED == hass.bus.listen.call_args_list[0][0][0]
-    assert get_write_api(mock_client).call_count == 1
+    listener = await _setup(hass, config)
+    listener.shutdown()
```

Comment:
Switching to `async_setup` in `influxdb` caused the integration to start generating the `I/O detected in the event loop` warning. This is because the integration was doing a test write to Influx during setup. I removed that to eliminate this warning and adjusted error handling during batch writes to account for invalid config situations. But since tests like this one were validating that write was happening this meant they needed adjustment. I also felt it would be easier to refactor duplicated logic into a helper method.

Reply:
Why not just schedule the I/O on the executor thread pool?

Reply:
You mean like since `InfluxThread` makes its own loop run it there instead of in the hass loop? That's effectively what the code is doing now to be honest. Previously what happened is it only created the `InfluxThread` once it knew the connection was good otherwise it waits for a time and tries again. Now the code creates it immediately and lets it handle all external connections in its own loop.

The only significant difference here is just that I stopped doing a dummy write to validate the connection. In the current code to test a V1 connection it did `write_points([])` which effectively 'write nothing'. It worked as a test but it was odd and I didn't say any documentation recommending a connection be tested that way in the python library. In V2 there was actually no equivalent to this. If you notice in the removed code it was doing `.write_api().write([])` then checking to see if the exception was status code 400 (meaning connection is good but inputs are invalid) or any other status code (meaning connection is bad). I really didn't like relying on an exception in the success code.

Now it just starts the thread and starts recording right away. If it gets an error on write it handles it based on the type of error:

1. If its invalid inputs - log the error only. Nothing can be done to fix this just let the user know
2. If its a connection issue - log the error and put the thread to sleep for an interval to give the user a chance to bring the Influx system back up if it was down, change the access on the token/user, etc. Or shutdown if they gave bad info

To me this was the same thing as before except testing the connection using real writes in `InfluxThread` instead of dummy writes during `setup_component`. Plus even if that dummy write succeeded Influx may go down later, better to handle connection errors all in one place.

Reply:
The point of doing the test in the component set up is to be able to fail the set up if there's a non recoverable error. We want to keep that functionality.

Reply:
Well actually that's not exactly what it is doing today.  The [current code](https://github.com/home-assistant/core/blob/dev/homeassistant/components/influxdb/__init__.py#L273) is doing this:
```python
    try:
        influx = get_influx_connection(kwargs, bucket)
        if use_v2_api:
            write_api = influx.write_api(write_options=ASYNCHRONOUS)
    except (exceptions.InfluxDBClientError, requests.exceptions.ConnectionError) as exc:
        _LOGGER.error("...Long error message...")
        event_helper.call_later(hass, RETRY_INTERVAL, lambda _: setup(hass, config))
        return True
    except ApiException as exc:
        _LOGGER.error("...Long error meessage...")
        event_helper.call_later(hass, RETRY_INTERVAL, lambda _: setup(hass, config))
        return True
```
So as you can see, no matter what its returning `True`. It never fails the setup even if the connection is bad. It just schedules setup to try again after a time. The only time setup actually fails is if the config is invalid, if the config is valid then a failed connection won't stop it from saying its set up.

What you're saying does make sense though. Is there a way to schedule retries like this and let HA know the actual status of the component during this setup? That although setup failed initially it succeeded later on a retry?

Reply:
If there's a non recoverable error, eg bad credentials, we should fail the setup. If there's a recoverable error we can either retry the setup as is done today or further down the stream.

In the config entries api we can raise `ConfigEntryNotReady`. That's not available in the legacy component setup api.

Reply:
Ok that seems reasonable. And yea I am hoping to make this component UI configurable and use config entries but one PR at a time.

So I'll go back to testing the setup and fail setup if the credentials or connection info is bad. Btw, is it possible to do this as a custom validator or is that a bad place to do I/O?

Also one other note, if setup of the component fails the sensor setup will also fail. There's no actual connection between the two, sensor specifies is own connection details. But because its all one platform sensor setup can only succeed if the platform was setup successfully.